app/uk/gov/hmrc/customs/declarations/information/services/declarationsInformationServices.scala

```diff
@@ -200,36 +200,44 @@ abstract class DeclarationService @Inject()(override val apiSubFieldsConnector:
-            logger.warn(s"declaration [$endpointName] call failed with backend http status code of 403: [${e.getMessage}] so returning 500 to consumer")
-            Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
-          case e: HttpException if e.responseCode == NOT_FOUND =>
-            logger.warn(s"declaration [$endpointName] call failed with backend http status code of 404: [${e.getMessage}] so returning 500 to consumer")
-            Left(customNotFoundResponse.XmlResult.withConversationId)
-          case e: HttpException =>
-            logger.warn(s"declaration [$endpointName] call failed with backend http status code of [${e.responseCode}]: [${e.getMessage}] so returning 500 to consumer")
-            Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
-          case _: CircuitBreakerOpenException =>
-            logger.error("unhealthy state entered so returning 500 to consumer with message service unavailble")
-            Left(errorResponseServiceUnavailable.XmlResult.withConversationId)
-          case NonFatal(e) =>
-            logger.error(s"declaration [$endpointName] call failed: [${e.getMessage}] so returning 500 to consumer", e)
-            Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
-        }
+          })
+          .recover(recoverException(asr, hc))
       case Left(result) =>
         Future.successful(Left(result))
     }
   }
 
+  private def recoverException[A](implicit asr: AuthorisedRequest[A], hc: HeaderCarrier): PartialFunction[Throwable, Either[Result, HttpResponse]] = {
+    case e: Non2xxResponseException if e.responseCode == INTERNAL_SERVER_ERROR =>
+      val body = XML.loadString(e.response.body)
+      val errorCode: NodeSeq = body \ "errorCode"
+      val errorCodeText = errorCode.text
+      matchErrorCode(errorCodeText)
+    case e: Non2xxResponseException if e.responseCode == FORBIDDEN && config.informationConfig.payloadForbiddenEnabled =>
+      returnErrorResult(asr, e, 403, ErrorResponse.ErrorPayloadForbidden)
+    case e: Non2xxResponseException if e.responseCode == FORBIDDEN =>
+      return500ErrorResult(asr, e)
+    case e: HttpException if e.responseCode == NOT_FOUND =>
+      returnErrorResult(asr, e, 500, customNotFoundResponse)
+    case e: HttpException =>
+      return500ErrorResult(asr, e)
+    case _: CircuitBreakerOpenException =>
+      logger.error("unhealthy state entered so returning 500 to consumer with message service unavailble")
+      Left(errorResponseServiceUnavailable.XmlResult.withConversationId)
+    case NonFatal(e) =>
+      logger.error(s"declaration [$endpointName] call failed: [${e.getMessage}] so returning 500 to consumer", e)
+      Left(ErrorResponse.ErrorInternalServerError.XmlResult.withConversationId)
+  }
+
+  private def return500ErrorResult[A](implicit asr: AuthorisedRequest[A], e: HttpException): Left[Result, Nothing] = {
```

Comment:
What makes this necessary? I think it would be clearer, visually and logically speaking, if we used `returnErrorResult` for all the cases.

Reply:
@kevin-co-hmrc this method returns 500 error message regardless of what response code.

Reply:
I must say I agree with Kevin here although it's not a huge thing. I think that returnErrorResult will do the exact same thing (unless you want to return 500 for 502s etc). But if you do want to do that then think would fit best within the same method. It is a simpler way of doing it without adding a slightly confusing bit of code. Overall good job though looks good :)

Reply:
As you can see that 403 is wanting to return 500 and same for the HttpException case that any response code doesn't match the previous condition will need to return 500. Happy to take it out if you really prefer the one below.
So here will be like if I don't have return500ErrorResult.
<img width="1206" alt="image" src="https://github.com/hmrc/customs-declarations-information/assets/132441448/90154b5c-800b-4905-90e5-86130cf9ea10">

Reply:
I can see that having the extra method makes it more apparent that the mismatch between the `e.responseCode` in the case clause guard and the actual response code being returned is entirely intentional.

I'm happy either way.